openedx/core/djangoapps/oauth_dispatch/docs/decisions/0006-enforce-scopes-in-LMS-APIs.rst

```diff
@@ -0,0 +1,225 @@
+6. Enforce Scopes in LMS APIs
+-----------------------------
+
+Status
+------
+
+Proposed
+
+Context
+-------
+
+Although external edX clients, as Restricted Applications, can use edX
+as an Identity Provider, they cannot make any API calls on behalf of 
+users. In addition, server-to-server calls via the Client Credentials
+grant type is also limiting as our API endpoints do not allow organizations
+to access data for their own users. The lack of OAuth2 Scopes enforcement
+by our API endpoints prevents us from lifting these constraints.
+
+For additional background information on the current implementation,
+see the README_.
+
+.. _README: ../README.rst
+
+Decisions
+---------
+
+Add support for enforcing OAuth2 scopes by making the following advancements simultaneously.
+
+1. Define and configure new OAuth2 Scopes for accessing API resources
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* For now, we will start with an initial set of OAuth2 Scopes based on
+  immediate API needs.
+
+* OAuth2 clients should be frugal about limiting the scopes they request
+  in order to:
+
+  * keep the data payload small
+  * keep the UX of the user approval form reasonable
+  * follow principle of least priviledge
+
+2. Restricted Applications receive *unexpired* JWTs, signed with a *new key*
```

Comment:
Reminder: Use the term "Unrestricted Tokens" versus "Restricted Tokens" to distinguish between the 2 key types.